dashboard/src/shared/AppRepository.ts

```diff
@@ -47,6 +47,14 @@ export class AppRepository {
     return data;
   }
 
+  public static async validate(repoURL: string, authHeader: string, customCA: string) {
+    const { data } = await axiosWithAuth.post<ICreateAppRepositoryResponse>(
+      url.backend.apprepositories.validate(),
+      { appRepository: { name, repoURL, authHeader, customCA } },
+    );
+    return data;
```

Comment:
So I assume from code above that if this doesn't validate, it errors rather than returning a 200 with the validation failures? Right, checking the handler I see that is the case. I missed that when reviewing the backend. It means our API doesn't differentiate between its own internal errors and an actual request to the repo, nor to be able to return individual field errors later  (such as malformed URL vs missing credentials). Or as another eg, if the repo validation request to the actual repository was returned 500, we'd return that same status code as well as whatever body they send in the response, I think, rather than returning that the request was processed successfully by us, but the validation failed. Anyway, that's related to the backend implementation, not this, but see what you think.

Reply:
I am not sure I follow, there are two cases here:
 - We are not able to reach the target with the given input. Eg the URL certificate is invalid. In this case we return a 500 because we don't know if the error is related to the service not having the default CA or it's missing a custom CA in the request.
 - We reach the target. In this case, whatever the server returns, we proxy it to the user.

Reply:
> Apart from maybe changing the error code from 500 to something else in the first case, I am not sure what could be an improvement?

Right, it's the 500 we return when we didn't really have an internal server error that makes it obvious in my pov, but more generally, our API should return meaningful status codes for it's own service. So in the latter case that you mention where we reach the target and the validation fails (say incorrect creds), then the request to perform the validation succeeded (200) and should have the results of the validation. If we return a 403/401 it implies that the request to our service failed for that reason.

I'd just expected the return status from our API to be separate from the return status of the 3rd party so that the caller can differentiate between the request to our own API failing vs the request from our service to the external URI failing. Not a big deal, it was just a surprise that I should have noticed when reviewing the backend change.